apps/src/studio/StudioAnimation.js

```diff
@@ -176,32 +241,65 @@ StudioAnimation.prototype.redrawCenteredAt = function (center) {
  * Sets which animation to play out of the sprite sheet.
  * Animations are indexed by their position in the sprite sheet, where each
  * animation is its own column and animation zero is the far-left column.
+ * @param {!string} animationType
  * @param {!number} animationIndex
  */
-StudioAnimation.prototype.setCurrentAnimation = function (animationIndex) {
-  this.currentAnimation_ = animationIndex;
+StudioAnimation.prototype.setCurrentAnimation = function (animationType, animationIndex) {
+  this.currentAnimationType_ = animationType;
+  this.currentAnimationIndex_ = animationIndex;
 };
 
 /**
- * Set the animation rate for this item's sprite.
- * @param {number} framesPerSecond
+ * Creates a new special animation types based on specific frames to play from
+ * the sprite sheet.
```

Comment:
I'm not sure what a "special" animation is.  Can we give a better description of a use case here, or a more descriptive name?

Also, is there a reason StudioAnimation is responsible for tracking these, instead of StudioSpriteSheet storing the knowledge of what animations it contains?